packages/workspace/src/browser/workspace-service.ts

```diff
@@ -88,78 +83,103 @@ export class WorkspaceService implements FrontendApplicationContribution {
         return this.onWorkspaceChangeEmitter.event;
     }
 
+    protected async onWorkspaceFileChanged(newWorkspaceFile?: FileStat): Promise<void> {
```

Comment:
let's call it `updateWorkpaceFile` or `setWorkspaceFile` and with `newWorkspaceFile: FileStat | undefined`  that one passes `undefined` explicitly